src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin.cxx

```diff
@@ -826,23 +752,23 @@ actual_compute_subset_gradient_without_penalty(TargetT& gradient,
         VectorWithOffset<ListModeData::SavedPosition>
                 frame_start_positions(1, static_cast<int>(this->frame_defs.get_num_frames()));
 
-        long int more_events =
-                this->do_time_frame? 1 : this->num_events_to_use;
-
-        while (more_events)
-       {
+        while (true)
+          {
 
            if (this->list_mode_data_sptr->get_next_record(record) == Succeeded::no)
            {
                info("End of file!");
                break; //get out of while loop
            }
 
-           if(record.is_time() && end_time > 0.01)
+           if(record.is_time())
            {
-               current_time = record.time().get_time_in_secs();
+               //NE: multiplication with 0.001 should not be nessesary.
+               current_time = record.time().get_time_in_secs()*0.001;
```

Comment:
@NikEfth we cannot do this. You might have a problem in your listmode reader, but with this modification we would break all others. I will revert.

Reply:
I've removed this